cuqi/samples/_samples.py

```diff
@@ -190,6 +194,24 @@ def geometry(self):
             self._geometry = _DefaultGeometry(grid=np.prod(self.samples.shape[:-1]))
         return self._geometry
 
+    @property
+    def funvals(self):
+        """If `self.is_par` is True, returns a new Samples object of sample function values by applying :meth:`self.geometry.par2fun` on each sample. Otherwise, returns the Samples object itself."""
+        if self.is_par is True:
+            _funvals = np.empty((self.geometry.fun_dim, self.Ns))
+            for i, s in enumerate(self):
+                _funvals[:, i] = self.geometry.par2fun(s)
+            return Samples(_funvals, is_par=False, geometry=self.geometry)
+        else:
+            return self
+
+    @property
+    def geometry_dim(self):
+        if self.is_par:
+            return self.geometry.dim
+        else:
+            return self.geometry.fun_dim
```

Comment:
Do we really need this method? At least I do not think it should be public. If it is used in various places we can make it private. Also is it really the `dim` we are looking for or the `shape`?

Reply:
I made it hidden for now.